File: Programmers/Lee-Sang-Hoon/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;

        Set<Integer> set = new HashSet<>(); // 중복 제거를 위한 Set
        for(int num : nums) {
            set.add(num);
        }

        int max = nums.length / 2; // 최대로 가져갈 수 있는 폰켓몬 수

        answer = Math.min(set.size(), max); // 중복 제거한 폰켓몬 수와 최대로 가져갈 수 있는 폰켓몬 수 중 작은 값이 정답

        return answer;
    }
}

File: Programmers/Yoo-Seung-Hui/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    answer = 0
    
    # 중복을 제거한 list = 포켓몬 종류의 개수
    s = list(set(nums))
    

    # 가져갈 수 있는 포켓몬 수 (len(nums)//2) 보다 포켓몬 종류의 수가 많으면 :
    # 가져갈 수 있는 포켓몬 수가 최대 종류의 개수
    if len(s) >= len(nums)//2 :
        return len(nums) // 2
    # 반대로 적으면
    # 최대 종류의 수만 리턴
    else :
        return len(s)
    
    #  다른 사람 풀이 
    #  return min(len(ls)/2, len(set(ls)))
    #  반복문 말고 min()으로 더 작은 값 리턴
    
